watcher/local/watcher.go

```diff
@@ -15,12 +15,92 @@
 package local
 
 import (
+	"context"
+	"sync"
+
+	"github.com/pkg/errors"
+
 	"perun.network/go-perun/channel"
+	"perun.network/go-perun/watcher"
 )
 
 type (
+	// Watcher implements a local watcher.
+	Watcher struct {
+		rs channel.RegisterSubscriber
+		*registry
+	}
+
+	// Config represents the configuration for initializing a local watcher.
```

Comment:
For now, this will contain only one parameter. But, soon we will be extending it (when introducing the concept of "sync interval").

Reply:
By soon, you mean "in one of the next PR's" or "at an undefined point in the future"?

It would be easier to review if a PR is self consistent. Hence, introduce the Config struct when you actually need it. One variable is not enough to justify a config struct, I think.

Reply:
It is "at an undefined point in the future". I have removed the config type and directly used `RegisterSubscriber` as the function parameter.

Reply:
See 8d5ff4